index.js

```diff
@@ -33,6 +45,16 @@ function resolvePackage(parsed, options) {
     pkgRoot = path.dirname(pkgPath);
   }
 
+  if (currentPkg.name === parsed.package) {
+    if (isV1Addon(currentPkg)) {
+      return `${pkgRoot}/addon/${parsed.path}`;
+    } else if (isEmberApp(currentPkg)) {
+      return `${pkgRoot}/app/${parsed.path}`;
```

Comment:
Not sure how best to deal with this, but it seems that we can't actually just assume everything under the apps namespace is in `app/`.

For example, it is common for an apps tests to import from `my-app-name/tests/helpers` (referencing `tests/helpers/index.js`) from somewhere in `tests/` folder.

The same thing is true for importing from an addons test support code (referencing `addon-name/test-support` would actually need to resolve to `node_modules/addon-name/addon-test-support/index.js`).

Another example is importing from `my-app-name/config/environment`, this is actually resolved from `config/environment.js`.

Reply:
@rwjblue is `config` special, or `config/environment` specifically?  i.e. what would `my-app/config/uwot` resolve to?